File: sdks/python/examples/smoke.py
from arbiter_sdk import ArbiterClient

SOURCE = """
rule Approve {
    when { user.score >= 700 }
    then Approved { tier: "gold" }
}
"""


def main() -> None:
    with ArbiterClient("127.0.0.1:18081") as client:
        publish = client.publish_bundle("python-smoke", SOURCE)
        result = client.evaluate_rules(
            bundle_name="python-smoke",
            context={"user": {"score": 720}},
            request_id="py-smoke",
        )
        if len(result.matched) != 1 or result.matched[0].action != "Approved":
            raise SystemExit(f"unexpected evaluate response: {result}")
        print(f"python sdk ok: {publish.bundle_id}")

File: sdks/node/examples/smoke.js
const { ArbiterClient } = require("../src");

const source = `
rule Approve {
  when { user.score >= 700 }
  then Approved { tier: "gold" }
}
`;

async function main() {
  const client = new ArbiterClient("127.0.0.1:18081");
  try {
    const publish = await client.publishBundle({ name: "node-smoke", source });
    const result = await client.evaluateRules({
      bundleName: "node-smoke",
      context: { user: { score: 720 } },
      requestId: "node-smoke",
    });
    if (!result.matched || result.matched.length !== 1 || result.matched[0].action !== "Approved") {
      throw new Error(`unexpected evaluate response: ${JSON.stringify(result)}`);
    }
    console.log(`node sdk ok: ${publish.bundleId}`);
  } finally {
    client.close();
  }
}

main().catch(err => {
  console.error(err);
  process.exit(1);
});
